tensorflow/python/eager/context_test.py

```diff
@@ -131,6 +131,12 @@ def testGetMemoryUsageAmbiguousDevice(self):
     with self.assertRaisesRegex(ValueError, 'Multiple devices'):
       context.context().get_total_memory_usage('GPU')
 
+  def testParsePhysicalDevices(self):
+    expected_cpu = context.PhysicalDevice(
+                     name='/physical_device:CPU:0',
+                     device_type='CPU', subdevice_type='CPU')
+    cpu_devices = context.context().list_physical_devices("CPU")
+    self.assertTrue(expected_cpu in cpu_devices)
```

Comment:
Also add a test for the case when subdevice type is defined ('/physical_device:GPU:X_GPU:0')

Reply:
This case is following current config_test: [testPhysicalDevice](https://github.com/tensorflow/tensorflow/blob/89c2f7f062ee125bd3783035b1be07b5c0f9ea73/tensorflow/python/framework/config_test.py#L372), because device list is dynamic queried, we can only make sure that cpu device is always exist

Reply:
Let's use mock to add a test for a different subdevice type:
```python
from tensorflow.python import pywrap_tfe

...

@test.mock.patch.object(pywrap_tfe, 'TF_ListPhysicalDevices')
def testParsePhysicalDeviceWithSubdevice(self, mock_list_devices):
    mock_list_devices.return_value = [b'/physical_device:GPU:XPU:0']
    new_context = context.Context()
    expected_gpu = context.PhysicalDevice(
        name='/physical_device:GPU:XPU:0',
        device_type='GPU',
        subdevice_type='XPU')
    gpu_devices = new_context.list_physical_devices('GPU')
    self.assertListEqual(gpu_devices, [expected_gpu])
    xpu_devices = new_context.list_physical_devices('XPU')
    self.assertListEqual(xpu_devices, [])
```

Reply:
done